commands/build_test.go

```diff
@@ -0,0 +1,159 @@
+package commands_test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpack/pack"
+	"github.com/buildpack/pack/commands"
+	cmdmocks "github.com/buildpack/pack/commands/mocks"
+	"github.com/buildpack/pack/config"
+	"github.com/buildpack/pack/internal/fakes"
+	"github.com/buildpack/pack/logging"
+	h "github.com/buildpack/pack/testhelpers"
+)
+
+func TestBuildCommand(t *testing.T) {
+	spec.Run(t, "Commands", testBuildCommand, spec.Random(), spec.Report(report.Terminal{}))
+}
+
+func testBuildCommand(t *testing.T, when spec.G, it spec.S) {
+	var (
+		command        *cobra.Command
+		logger         logging.Logger
+		outBuf         bytes.Buffer
+		mockController *gomock.Controller
+		mockClient     *cmdmocks.MockPackClient
+		cfg            config.Config
+	)
+
+	it.Before(func() {
+		logger = fakes.NewFakeLogger(&outBuf)
+		cfg = config.Config{}
+		mockController = gomock.NewController(t)
+		mockClient = cmdmocks.NewMockPackClient(mockController)
+
+		command = commands.Build(logger, cfg, mockClient)
+	})
+
+	when("#BuildCommand", func() {
+		when("no env file is provided", func() {
+			it("builds an image with a builder", func() {
+				mockClient.EXPECT().
+					Build(gomock.Any(), EqBuildOptionsWithImage("my-builder", "image")).
+					Return(nil)
+
+				command.SetArgs([]string{"--builder", "my-builder", "image"})
+				h.AssertNil(t, command.Execute())
+			})
+		})
+
+		when("an env file is provided", func() {
+			var envPath string
+
+			it.Before(func() {
+				envfile, err := ioutil.TempFile("", "envfile")
+				h.AssertNil(t, err)
+				defer envfile.Close()
+
+				envfile.WriteString(`KEY=VALUE`)
+				envPath = envfile.Name()
+			})
+
+			it("builds an image with a builder", func() {
```

Comment:
It would be better if the description strings in each `it` described what this particular test was asserting on (i.e. env vars in this case)